src/ripple/app/paths/PathRequest.cpp

```diff
@@ -307,8 +307,7 @@ int PathRequest::parseJson (Json::Value const& jvParams)
         return PFR_PJ_INVALID;
     }
 
-    convert_all_ =
-        saDstAmount == STAmount(saDstAmount.issue(), 1u, 0, true);
+    convert_all_ = saDstAmount == STAmount(saDstAmount.issue(), 1u, 0, true);
```

Comment:
> Also note the comparison is against -1 not 1.

Hah!  :-)

And in fact, I "knew" that - I spent a long time talking to Mickey about it the last time this came up - and yet it was so easy for me to look right at the expression `STAmount(saDstAmount.issue(), 1u, 0, true)` and call it `+1` when talking about it...